fulltext/backends/__json.py

```diff
@@ -0,0 +1,32 @@
+import json
+
+from six import StringIO
+from six import string_types
+from six import integer_types
+
+
+def _to_text(text, obj):
+    if isinstance(obj, dict):
+        for key in sorted(obj.keys()):
+            _to_text(text, key)
+            _to_text(text, obj[key])
+
+    elif isinstance(obj, list):
+        for item in obj:
+            _to_text(text, item)
+
+    elif isinstance(obj, string_types):
+        text.write(obj)
+        text.write(u' ')
```

Comment:
maybe do `obj + u' '`? (faster)

Reply:
Format strings are far faster than concatenation. `StringIO().write()` is in the middle. I just benchmarked this when optimizing duster (making shingles).